editor/test/integration/test_util.clj

```diff
@@ -793,33 +793,34 @@
     handler/selection))
 
 ;; Extension library server
-
-(defn ->lib-server []
-  (doto (http-server/->server 0 {"/lib" (fn [request]
-                                          (let [lib (subs (:url request) 5)
-                                                path-offset (count (format "test/resources/%s/" lib))
-                                                ignored #{".internal" "build"}
-                                                file-filter (reify FilenameFilter
-                                                              (accept [this file name] (not (contains? ignored name))))
-                                                files (->> (tree-seq (fn [^File f] (.isDirectory f)) (fn [^File f] (.listFiles f file-filter)) (io/file (format "test/resources/%s" lib)))
-                                                        (filter (fn [^File f] (not (.isDirectory f)))))]
-                                            (with-open [byte-stream (ByteArrayOutputStream.)
-                                                        out (ZipOutputStream. byte-stream)]
-                                              (doseq [^File f files]
-                                                (with-open [in (FileInputStream. f)]
-                                                  (let [entry (doto (ZipEntry. (subs (.getPath f) path-offset))
-                                                                (.setSize (.length f)))]
-                                                    (.putNextEntry out entry)
-                                                    (IOUtils/copy in out)
-                                                    (.closeEntry out))))
-                                              (.finish out)
-                                              (let [bytes (.toByteArray byte-stream)]
-                                                {:headers {"ETag" "tag"}
-                                                 :body bytes}))))})
-    (http-server/start!)))
-
-(defn kill-lib-server [server]
-  (http-server/stop! server))
+(defmacro with-server [handler & body]
```

Comment:
Rename to `with-http-server`? This is a bit generic for `test-util`. There can be other servers eventually, I guess?

Reply:
Heh, I thought about it but didn't come up with other servers we might have. Maybe we could do it when we actually have another non-http server? 😄

Reply:
We could, but I prefer we do it now for clarity. We're a desktop application, so "server" could mean other things (a build server or native extension server maybe?).

I'd especially like to address this now since it has an entry in `codeStyles/Project.xml`, so a refactoring rename will mess up the indentation if we're not careful.